game-core/src/test/resources/Total_World_War_Dec1941.xml

```diff
@@ -7749,22 +7809,32 @@
       <option name="consumesUnits" value="1:Material"/>
       <option name="requiresUnits" value="germanCombatEngineer"/>
     </attachment>
-    <attachment name="unitAttachment" attachTo="germanInfantry" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
+    <attachment foreach="Infantry:Players:Barracks" name="unitAttachment" attachTo="$Infantry$" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
       <option name="movement" value="1"/>
       <option name="attack" value="2"/>
       <option name="defense" value="3"/>
       <option name="transportCost" value="2"/>
-      <option name="isInfantry" value="true"/>
+      <option name="isLandTransportable" value="true"/>
+      <option name="canBeGivenByTerritoryTo" value="$Players$"/>
+      <option name="requiresUnits" value="$Barracks$"/>
+      <option name="canInvadeOnlyFrom" value="none"/>
+    </attachment>
+    <!-- <attachment name="unitAttachment" attachTo="germanInfantry" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
```

Comment:
Why is this XML block commented? As a reader it raises a number of questions. It's very questionable how often commented out XML is useful vs invalid and out of date.

Reply:
Just left it as an example of what is getting generated with the new code that replaced it. Fine removing it if you don't think its helpful.

Reply:
Done.